janusgraph-core/src/main/java/org/janusgraph/graphdb/query/vertex/MultiVertexCentricQueryBuilder.java

```diff
@@ -202,4 +203,19 @@ public Map<JanusGraphVertex, VertexList> vertexIds() {
         return execute(RelationCategory.EDGE, new VertexIdConstructor());
     }
 
+    @Override
+    public Integer drop() {
+        Map<JanusGraphVertex, Iterable<JanusGraphRelation>> relations = this.noPartitionRestriction().relations();
+        Map<JanusGraphVertex, Iterable<JanusGraphRelation>> systemRelations = this.noPartitionRestriction().system().relations();
+
+        return relations
+            .keySet()
+            .stream()
+            .mapToInt(vertex -> {
+                Iterable<JanusGraphRelation> allRelations = Iterables.concat(relations.get(vertex), systemRelations.get(vertex));
+                ((InternalVertex) vertex).remove(allRelations);
+                return 1;
+            })
+            .sum();
```

Comment:
perf nit: this seems unnecessary computation from performance perspective. I would guess JVM will be able to optimize this, but I'd prefer not to rely on that. We could simply return `relations.size()` without doing summation.

Reply:
Yeah, seems can be just:
```
relations.keySet().foreach(vertex -> ((InternalVertex) vertex).remove(Iterables.concat(relations.get(vertex), systemRelations.get(vertex))));
return relations.size();
```

Reply:
Fixed, thanks @li-boxuan